crates/solver/src/driver.rs

```diff
@@ -233,6 +235,17 @@ impl Driver {
         Ok(())
     }
 
+    fn observe_auction_orders(&mut self, orders: &[Order]) {
+        let orders: HashSet<OrderUid> = orders.iter().map(|order| order.metadata.uid).collect();
+        for order in orders.difference(&self.previous_auction_orders) {
+            tracing::debug!(%order, "new order in auction");
```

Comment:
nit: The same order may drop in and out of auctions so `new order` might be slightly misleading.
```suggestion
            tracing::debug!(%order, "order now in auction");
```

Reply:
As a side note: did you consider logging only a single message containing all ids? It seems like having 1 log per order might be a bit more noisy than needed (especially during start ups where all orders in the auction will be printed).

Reply:
Disabled auto merge while discussing.

I'll change the log message to be more clear about "new".

>  logging only a single message

Either way the amount of information is the same but the number of log messages changes. Doing it this way felt more natural to me because the log message is about individual orders and not about all of them as a set. I care about seeing that some specific order is in the auction. I don't care about seeing that these orders showed up together.

Reply:
> But if you're looking at all the logs of an auction then maybe logging once is nicer.

Yeah, I was thinking about that. However, this is probably not so big of an issue so I'm also fine with having 1 log per order.

Reply:
I decided I like merged logs more.